doc/contributing.md

```diff
@@ -0,0 +1,247 @@
+for the graders. You probably shouldn't do any work here, but rather should
+explore in other locations such as scrap paper. Once you have several solutions,
+you can pick the one you like the best, and more neatly copy the work on to the
+test you will hand in.
+
+Our master branch can be found [here](https://github.com/RoboJackets/robocup-software).
+The code in latest master is always neat and untouched. It can always compile. When you
+first clone our codebase from git to view the simulator and soccer, you are
+using the code in master. It is in all respects, the master copy from which all
+other contributions are derived. Even if you are eventually given permission to
+write to master, you should never do so.
+
+### Forks
+If the professor puts test questions on the whiteboard for the whole class, you
+would not be allowed to solve the problems on the board; you would be expected
+to copy any relevant information to your own paper and solve the problems there.
+This gives you the freedom to play around with rephrasing and solving questions,
+without distuirbing others.
+
+Forking a repository on Github duplicates the project, but you are given full
+write access to your own duplicate. Now you can delete, recreate, and add code
+relevent to your contribution without harming the progress of others. This 
+duplicate is known as your repository and every team member has a fork. This is 
+different from the main repository which belongs to the team rather than an 
+individual. Don't confuse the main repository with the master branch. The
+shared, main repository can have incomplete features being worked on by 
+everyone. The master in your fork and the master in the main repository should
+always remain clean.
+
+This also works well when you want a feature that only some people want. You 
+don't have to move every contribution from your fork back into the main 
+repository. When you have contributed something and want it placed in master, 
+whoever is reviewing your code will look at your fork of the code and compare 
+it to the master branch.
+
+### Branches
+Branches are like pieces of scrap paper. You can use them to organize your work 
+and solutions to the test questions. You should not have work regarding 
+different problems mixed across several pieces of paper; you may get your 
+progress confused. You should use one (or several) peices of paper for each
+problem you are trying to solve, but should never use one sheet for multiple
+questions.
+
+For RoboCup you should create a new branch for every new item you'd like to work
+on and for every bug or issue you have to fix. This ensures your master branch
+stays clean. You should never solve more than one issue at a time and you should
+never have changes or additions for multiple things in the same branch. You can 
+look at a typical branching layout ![branchingModel](http://justinhileman.info/article/changing-history/git-flow.png).
```

Comment:
Not sure why, but GitHub displays the markdown for this with "layout" below the picture... If you put the image on a new line below the paragraph it should fix it.